storage/innobase/btr/btr0cur.cc

```diff
@@ -413,6 +406,13 @@ btr_cur_latch_leaves(
 
 		latch_leaves.blocks[1] = get_block;
 #ifdef UNIV_BTR_DEBUG
+		/* Sanity check only after both the blocks are latched. */
+		if (latch_leaves.blocks[0] != NULL) {
+			ut_a(page_is_comp(latch_leaves.blocks[0]->frame)
+			     == page_get_page_no(page));
+			ut_a(btr_page_get_next(latch_leaves.blocks[0]->frame, mtr)
+			     == page_get_page_no(page));
+		}
```

Comment:
1. How is this found? any testcase? because this should happen only special builds when compiled with UNIV_BTR_DEBUG (I think debug builds dont enable this flag by default)

2. How do we know the page is locked here? Can we add the assert that page is locked? 
 mtr_memo_contains_flagged(
              mtr, dict_index_get_lock(index),
              MTR_MEMO_S_LOCK | MTR_MEMO_X_LOCK | MTR_MEMO_SX_LOCK)

Use appropriate lock mode here

Reply:
HI @satya-bodapati 
1. How is this found? - This is found by customer, reproducible only by customer. Fix was proposed in PR #3862 but without respect to GCA process. As explained the fix was based on identical fix from Oracle done in commit 777bf2a but for another case in the same switch (a few lines above)
UNIV_BTR_DEBUG is defined in univ.i unconditionally.
2. How do we know the page is locked here? - btr_block_get() locks the page

The change was validated by customer (and still is under validation). Please see PS-7010 for details.

Reply:
@kamil-holubicki 

UNIV_BTR_DEBUG is undefined. So enable this, either we modify the univ.i file or pass to cmake cxx/c flags .

Ok, is it possible to add the assertion?

Can you also compile with UNIV_BTR_DEBUG and see if this code is reached via some testcase?

Reply:
may be add ut_a(0), enable UNIV_BTR_DEBUG and run innodb mtr suite?